src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -14,6 +15,30 @@
 
 namespace Dynamo.Configuration
 {
+    static class ExtensionMethods
+    {
+        /// <summary>
+        /// Copy Properties from a PreferenceSettings instance to another
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="destination"></param>
+        internal static void CopyProperties(this PreferenceSettings source, PreferenceSettings destination)
+        {
+            // Iterate the Properties of the destination instance and populate them from their source counterparts  
+            PropertyInfo[] destinationProperties = destination.GetType().GetProperties();
+
+            foreach (PropertyInfo destinationPi in destinationProperties)
+            {
+                PropertyInfo sourcePi = source.GetType().GetProperty(destinationPi.Name);
+
+                if (destinationPi.GetCustomAttributes(typeof(System.ObsoleteAttribute), true).Length == 0 && destinationPi.CanWrite)
```

Comment:
so if a property is obsolete it's not copied? This should go in the description/summary of the method.

Reply:
if a property is obsolete it just will be skipped, both old and new instance are created from the same class (defined property), I will create a test for it.